util/os-resources.py

```diff
@@ -0,0 +1,84 @@
+# !/usr/bin/python3
+import click
+import shutil
+import os
+
+@click.group()
+def cli():
+    pass
+
+@cli.command()
+@click.option("--cookbooks-path", help="Directory containing the cookbooks.", required=True, type=click.Path())
+@click.option("--src-platform", help="Platform of the os to copy from (e.g. redhat)", required=True)
+@click.option("--src-version", help="Version of the os to copy from (e.g. 8)", required=True)
+@click.option("--dst-platform", help="Platform of the os to copy to (e.g. rocky)", required=True)
+@click.option("--dst-version", help="Version of the os to copy to (e.g. 8)", required=True)
+def generate(cookbooks_path, src_platform, src_version, dst_platform, dst_version):
+    src_os = f"{src_platform}{src_version}"
+
+    resources = _get_resources_by_os(cookbooks_path, src_os)
+    _generate_new_resources(resources, dst_platform, dst_version, src_platform, src_version)
+
+def _generate_new_resources(resources, dst_platform, dst_version, src_platform, src_version):
+    src_os = f"{src_platform}{src_version}"
+    dst_os = f"{dst_platform}{dst_version}"
+
+    for resource in resources:
+        dirname = os.path.dirname(resource)
+
+        src_filename = os.path.basename(resource)
+        src_path = os.path.join(dirname, src_filename)
+
+        dst_name = src_filename.replace(src_os, dst_os)
+        dst_path = os.path.join(dirname, dst_name)
+        _copy_and_replace(dst_path, dst_platform, dst_version, src_path, src_platform, src_version)
+
+def _copy_and_replace(dst_path, dst_platform, dst_version, src_path, src_platform, src_version):
+    print(f"Creating {dst_path}")
+    shutil.copy(src_path, dst_path)
+
+    with open(dst_path, 'r') as file:
+        filedata = file.read()
+        # Replace platform in the dst file
+        filedata = filedata.replace(f"platform: '{src_platform}'", f"platform: '{dst_platform}'")
+        filedata = filedata.replace(f"platform?('{src_platform}')", f"platform?('{dst_platform}')")
+        if f"{dst_platform}" not in filedata:
+            print(f"ERROR: Unable to replace platform in file {dst_path}")
+
+        # Replace platform in the dst file. There are 2 alternatives ways to find version
+        filedata = filedata.replace(f"to_i == {src_version}", f"to_i == {dst_version}")
+        filedata = filedata.replace(f"platform_version: '{src_version}'", f"platform_version: '{dst_version}'")
+
+        if f"to_i == {dst_version}" not in filedata and f"platform_version: '{dst_version}'" not in filedata:
```

Comment:
I think it's safer to have an or condition rather than an and condition because we want  both to be replaced.

Reply:
This check is for "version" only, that can be replaced with 2 different patterns, but only one of them is present in the files.